### PR TITLE
Add dependabot configuration

### PR DESCRIPTION
**Pull request by trask**
this is the same configuration as used in java-instrumentation repo

### CODE REVIEW

**Review by jeanbisutti — APPROVED** [review]

File: .github/dependabot.yml [hunk 1]
```diff
@@ -0,0 +1,20 @@
+version: 2
+registries:
+  gradle-plugin-portal:
+    type: maven-repository
+    url: https://plugins.gradle.org/m2
+    username: dummy # Required by dependabot
```
Comment (heyams) [inline .github/dependabot.yml:6]: dependabot needs to drop username and password from the yml. a feature ask for them? it's a weird setup and we need to suppress the dummy cred.

**Review by heyams — APPROVED** [review]

File: .github/dependabot.yml (hunk 1 above)
Comment (trask) [reply .github/dependabot.yml]: I didn't look into it, just copied this from OpenTelemetry 😄

I removed the creds, and will merge this and see what happens, and will bring them back if needed

Comment (trask) [reply .github/dependabot.yml]: ☹️

> Your .github/dependabot.yml contained invalid details
Dependabot encountered the following error when parsing your .github/dependabot.yml:
>
> The property '#/registries/gradle-plugin-portal/' did not contain a required property of 'username'
The property '#/registries/gradle-plugin-portal/' did not contain a required property of 'password'
Please update the config file to conform with Dependabot's specification.

but looks like an empty `""` password may work, curious if cred scanner will ignore that

Comment (heyams) [reply .github/dependabot.yml]: worth of trying :)

**Pull request merged**